### PR TITLE
Refactor list_cos_objects() w/ resource api

### PR DESCRIPTION
**Pull request by shawnzhu**
Supports #49 

Notice that this is a break change since the columns of the return value of `list_cos_objects()` has been changed. From `['ObjectURL', 'Size']` to `['bucket_name', 'key']`. No idea if this is acceptable.

### CODE REVIEW

File: Python/ibmcloudsql/SQLQuery.py [hunk 1]
```diff
@@ -555,34 +566,15 @@ def list_cos_objects(self, url):
...
+        objects = cos.Bucket(bucket).objects.filter(Prefix=prefix)
+        obj_summary_columns = ['bucket_name', 'key']
 
-        objects_df = pd.DataFrame(columns=['ObjectURL', 'Size'])
-        for object in bucket_objects:
-            objects_df = objects_df.append([{'ObjectURL': object['Key'], 'Size': object['Size']}], ignore_index=True)
-
-        return objects_df
+        return pd.DataFrame(
+            [[obj['bucket_name'], obj['key']] for obj in objects],
```
Comment (torsstei) [inline Python/ibmcloudsql/SQLQuery.py]: Thank you @shawnzhu 

Seems there's still a little glitch here:
```
Traceback (most recent call last):
  File "ibmcloudsql/test.py", line 155, in <module>
    objects_df = sqlClient.list_cos_objects(test_credentials.result_location)
  File "/Users/torsten@de.ibm.com/sql-query-clients/Python/ibmcloudsql/lib/python2.7/site-packages/ibmcloudsql/SQLQuery.py", line 575, in list_cos_objects
    [[obj['bucket_name'], obj['key']] for obj in objects],
TypeError: 's3.ObjectSummary' object has no attribute '__getitem__'
```

Comment (torsstei) [reply Python/ibmcloudsql/SQLQuery.py]: Also increase version in setup.py please.

**Comment by torsstei** [comment]
Is there a way to get the object sizes returned from. boto3 too?

File: Python/ibmcloudsql/SQLQuery.py (hunk 1 above)
Comment (shawnzhu) [reply Python/ibmcloudsql/SQLQuery.py]: fixed the glitch in https://github.com/IBM-Cloud/sql-query-clients/pull/50/commits/77bbd817d2ea076f19081162723c34005f44fe18

**Comment by shawnzhu** [comment]
> Is there a way to get the object sizes returned from. boto3 too?

Yes, but not via request `listObjects()`. see https://github.com/IBM/ibm-cos-sdk-python/blob/88551528ccc89625fea6df8887d17d00bb5de919/ibm_boto3/data/s3/2006-03-01/resources-1.json#L196-L210

I saw this test case indicates that it doesn't expose attribute `content_length` via `ibm_boto3`: https://github.com/IBM/ibm-cos-sdk-python/blob/014deaa8e7f0631df6561c5c3033ad1780b39fed/tests/functional/test_s3.py#L532-L535

**Comment by torsstei** [comment]
Thanks @shawnzhu.

Does `<boto3 client>.Bucket(<bucket>).objects.filter(Prefix=<prefix>)` return all objects or. does it also only get us the first 1000 objects? 

In any way, we should converge get_cos_summary() and list_cos_objects() to use the same method. The latter uses pagination and while paging through also gets the object sizes.

**Comment by torsstei** [comment]
OK, merging for now. But we need to converge get_cos_summary an d list_cos_objects code paths!!

**Pull request merged**